src/OpenTelemetry/Metrics/MetricReaderExt.cs

```diff
@@ -102,12 +111,16 @@ internal List<Metric> AddMetricsListWithViews(Instrument instrument, List<Metric
                         continue;
                     }
 
-                    if (this.metricStreamNames.Contains(metricStreamName))
+                    if (this.instrumentIdentityToMetric.TryGetValue(instrumentIdentity, out var existingMetric))
```

Comment:
we need to add `existingMetric` to the metrics.Add(existingMetric) as well.

Reply:
consider example:
instr1 has view, which says produce 2 streams named inst1_a, inst1_b.
instr2 has view, which says produce 2 streams named inst1_a, inst1_c.

instr1 creation will return List{M1,M2}.
instr2 creation *should* return List {M1,M3} --> this PR currently only returns {M3}.

^ we should add unittest to cover this.

Reply:
You are correct. Views hurt my mind 🤕. Fixed https://github.com/open-telemetry/opentelemetry-dotnet/pull/2916/commits/2dd0058ee90faaea17fcdecd606ca919145328f8

Reply:
And yet, everyone loves a room with a view! 😄